src/linter/rules/invalid-change-version.mjs

```diff
@@ -50,20 +59,40 @@ const isInvalid = NODE_RELEASED_VERSIONS
 /**
  * Identifies invalid change versions from metadata entries.
  *
- * @param {ApiDocMetadataEntry[]} entries - Metadata entries to check.
- * @returns {import('../types').LintIssue[]} List of detected lint issues.
+ * @param {import('../types.d.ts').LintContext} context
+ * @returns {void}
  */
-export const invalidChangeVersion = entries =>
-  entries.flatMap(({ changes, api_doc_source, yaml_position }) =>
-    changes.flatMap(({ version }) =>
-      (Array.isArray(version) ? version : [version])
+export const invalidChangeVersion = context => {
+  visit(context.tree, createQueries.UNIST.isYamlNode, node => {
+    const yamlContent = extractYamlContent(node);
+
+    const normalizedYaml = normalizeYamlSyntax(yamlContent);
+
+    // TODO: Use YAML AST to provide better issues positions
```

Comment:
DO you wanna resolve this todo in this pr or in follow up one ?

Reply:
I can work on a follow-up pr